.github/workflows/ci-tls.yml

```diff
@@ -0,0 +1,111 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This workflow claims the names of the unpublished crates in this repository
+# on crates.io (by publishing a dummy empty package)
+
+# Allow only one crate claim workflow to run at a time
+concurrency:
+  group: tls-configuration-smithy-rs
+  cancel-in-progress: true
+
+env:
+  rust_version: 1.68.2
+
+name: Verify client TLS configuration
+on:
+  pull_request:
+
+jobs:
+  verify-tls-config:
+    name: Verify TLS configuration
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install packages
+        shell: bash
+        run: |
+          sudo apt -y install gcc make python3-pip nginx git
+          pip3 install certbuilder crlbuilder
+      - name: Checkout smithy-rs
+        uses: actions/checkout@v3
+        with:
+          path: ./smithy-rs
+      - name: Checkout trytls
+        uses: actions/checkout@v3
+        with:
+          repository: ouspg/trytls
+          path: ./trytls
+      - name: Checkout badtls
+        uses: actions/checkout@v3
+        with:
+          repository: wbond/badtls.io
+          path: ./badtls.io
+      - name: Checkout badssl
+        uses: actions/checkout@v3
+        with:
+          repository: chromium/badssl.com
+          path: ./badssl.com
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.rust_version }}
+      - name: Build badssl.com
+        shell: bash
+        working-directory: badssl.com
+        env:
+          DOCKER_BUILDKIT: 1
+        run: |
+          perl -p -i -e 's/ruby2\.4/ruby2.6/' Dockerfile
+          grep -q 'start of badssl\.test hosts' /etc/hosts || make list-hosts | sudo tee -a /etc/hosts
+          make serve || true
+          ls certs/sets/test/gen/crt/ca-root.crt
+      - name: Build trytls
+        shell: bash
+        working-directory: trytls
+        run: |
+          perl -p -i -e 's!\./runners!runners!' setup.py
+          sed -i '/import platform/a import distro' runners/trytls/utils.py
+          sed -i 's/platform.linux_distribution()/distro.name(), distro.version(), distro.id()/' runners/trytls/utils.py
+          pip3 install -e .
+      - name: Build badtls.io
+        working-directory: badtls.io
+        shell: bash
+        run: |
+          python3 scripts/generate.py badtls.test
+          sudo mkdir /etc/nginx/tls || true
+          sudo mkdir /var/www || true
+          sudo python3 scripts/install.py /etc/nginx/conf.d /etc/nginx/tls /var/www
+          timeout 2m bash ./scripts/local.sh || true
```

Comment:
this can all be a separate `.sh` that you invoke, will simplify things a bit and make it easier to repro/debug